src/server/user_handler.cpp

```diff
@@ -49,3 +50,47 @@ std::string user_handler::create_salt(int length) {
 
 	return  ss.str();
 }
+
+// TODO - This really should be a common function but elsewhere it's a virtual member and I don't feel like fixing the mess tonight.
```

Comment:
Make sure to address the TODO before merging.

Reply:
That leads to the question of why it's a member function, much less a virtual member function. Which means rippling effects outward. With the tight coupling of Wesnoth code, who knows where that ends?

It's doable but, really, fixing this sort of crap seems out of scope for this change-set.

Reply:
won't fix

I need to create yet another file which is shared between wesnoth, wesnothd, and campaind. Then I need to compare all 3 (4? More?) versions of this function to be sure they're as identical as they look.

Too much work. If it's that important, someone else can do it for 1.15. I'm sure we have dozens of bits of code like this. (One analysis tool I've been playing with locally indicates there are several hundred potential, non-trivial duplicated blocks of code.)

Just not worth delaying 1.14 RC1 for.

Reply:
In that case, at least remove "I don't really feel like fixing this mess tonight" from the comment.